src/Compilers/CSharp/Portable/Errors/LazyMissingNonNullTypesContextDiagnosticInfo.cs

```diff
@@ -10,49 +11,69 @@ namespace Microsoft.CodeAnalysis.CSharp
     /// </summary>
     internal sealed class LazyMissingNonNullTypesContextDiagnosticInfo : LazyDiagnosticInfo
     {
-        private readonly CSharpCompilation _compilation;
-        private readonly bool _isNullableEnabled;
         private readonly TypeSymbolWithAnnotations _type;
+        private readonly DiagnosticInfo _info;
 
-        internal LazyMissingNonNullTypesContextDiagnosticInfo(CSharpCompilation compilation, bool isNullableEnabled, TypeSymbolWithAnnotations type)
+        private LazyMissingNonNullTypesContextDiagnosticInfo(TypeSymbolWithAnnotations type, DiagnosticInfo info)
         {
             Debug.Assert(!type.IsNull);
-            _compilation = compilation;
-            _isNullableEnabled = isNullableEnabled;
             _type = type;
+            _info = info;
         }
 
-        protected override DiagnosticInfo ResolveInfo()
+        public static void AddAll(bool isNullableEnabled, TypeSymbolWithAnnotations type, Location location, DiagnosticBag diagnostics)
         {
-            return ReportNullableReferenceTypesIfNeeded(_compilation, _isNullableEnabled, _type);
+            var rawInfos = ArrayBuilder<DiagnosticInfo>.GetInstance();
+            GetRawDiagnosticInfos(isNullableEnabled, (CSharpParseOptions)location.SourceTree.Options, rawInfos);
+            foreach (var rawInfo in rawInfos)
+            {
+                diagnostics.Add(new LazyMissingNonNullTypesContextDiagnosticInfo(type, rawInfo), location);
+            }
+            rawInfos.Free();
+        }
+
+        private static void GetRawDiagnosticInfos(bool isNullableEnabled, CSharpParseOptions options, ArrayBuilder<DiagnosticInfo> infos)
+        {
+            const MessageID featureId = MessageID.IDS_FeatureNullableReferenceTypes;
```

Comment:
Nit: not sure why this needs to be broken out for just a single usage.

Reply:
I found it more readable